src/seedu/addressbook/data/person/Address.java

```diff
@@ -6,27 +6,25 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact {
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
-    private boolean isPrivate;
 
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
+        super(address, isPrivate);
```

Comment:
is the stored value the same as before?

Reply:
The stored value should be `address.trim()`, which is an internal process under the parent class `Contact` that assigns `address.trim()` to the value attribute of an object using the line `this.value = address.trim()`.

Reply:
ahh I see, my bad. I think i probably raised the question because it can seem a little surprising that its value is stored without being trimmed but the trimmed value is checked for validity. 
Btw I just realised that you used `trimmedPhone` in `Contact`. I think the that is misleading

Reply:
Oops, I was careless there. Noted with thanks! :D